File: cf_sdk/src/main/java/com/example/cf_sdk/logic/changebankapi/network/api/ApiCreator.java
package com.example.cf_sdk.logic.changebankapi.network.api;

import com.example.cf_sdk.logic.changebanksdk.log.Logger;
import com.example.cf_sdk.logic.changebanksdk.util.Creator;

import retrofit2.Retrofit;

/**
 *
 * Interface to abstract the creation of an API.
 */

public interface ApiCreator<R> extends Creator<Retrofit, R>{
    @Override
    R create(Logger logger, Retrofit retrofit);
}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/network/api/ApiCreator.kt
package com.example.cf_sdk.changebankapi.network.api


import com.example.cf_sdk.changebankapi.log.Logger
import com.example.cf_sdk.changebankapi.util.Creator
import retrofit2.Retrofit

/**
 *
 * Interface to abstract the creation of an API.
 */
interface ApiCreator<R> : Creator<Retrofit?, R> {
    override fun create(logger: Logger?, parameter: Retrofit?): R
}